tests/test_scores.py

```diff
@@ -130,3 +132,51 @@ def test_scores_next():
         ("rrr", "qq", "rrrqq", 5)
     ]
     assert actual == expected
+
+
+def test_scores_by_referencey():
+    "Test scores_by_reference method."
+    spectrum_1 = Spectrum(mz=numpy.array([100, 150, 200.]),
+                          intensities=numpy.array([0.7, 0.2, 0.1]),
+                          metadata={'id': 'spectrum1'})
+    spectrum_2 = Spectrum(mz=numpy.array([100, 140, 190.]),
+                          intensities=numpy.array([0.4, 0.2, 0.1]),
+                          metadata={'id': 'spectrum2'})
+    spectrum_3 = Spectrum(mz=numpy.array([110, 140, 195.]),
+                          intensities=numpy.array([0.6, 0.2, 0.1]),
+                          metadata={'id': 'spectrum3'})
+    spectrum_4 = Spectrum(mz=numpy.array([100, 150, 200.]),
+                          intensities=numpy.array([0.6, 0.1, 0.6]),
+                          metadata={'id': 'spectrum4'})
+    references = [spectrum_1, spectrum_2, spectrum_3]
+    queries = [spectrum_3, spectrum_4]
+
+    scores = Scores(references, queries, CosineGreedy()).calculate()
+    selected_scores = scores.scores_by_reference(spectrum_2)
+
+    expected_result = [(scores.queries[:, i][0], scores.scores[1, i]) for i in range(2)]
+    assert selected_scores == expected_result, "Expected different scores."
+
+
+def test_scores_by_query():
+    "Test scores_by_query method."
+    spectrum_1 = Spectrum(mz=numpy.array([100, 150, 200.]),
+                          intensities=numpy.array([0.7, 0.2, 0.1]),
+                          metadata={'id': 'spectrum1'})
+    spectrum_2 = Spectrum(mz=numpy.array([100, 140, 190.]),
+                          intensities=numpy.array([0.4, 0.2, 0.1]),
+                          metadata={'id': 'spectrum2'})
+    spectrum_3 = Spectrum(mz=numpy.array([110, 140, 195.]),
+                          intensities=numpy.array([0.6, 0.2, 0.1]),
+                          metadata={'id': 'spectrum3'})
+    spectrum_4 = Spectrum(mz=numpy.array([100, 150, 200.]),
+                          intensities=numpy.array([0.6, 0.1, 0.6]),
+                          metadata={'id': 'spectrum4'})
+    references = [spectrum_1, spectrum_2, spectrum_3]
+    queries = [spectrum_2, spectrum_3, spectrum_4]
+
+    scores = Scores(references, queries, CosineGreedy()).calculate()
+    selected_scores = scores.scores_by_query(spectrum_4)
+
+    expected_result = [(scores.references[i][0], scores.scores[i, 2]) for i in range(3)]
```

Comment:
```suggestion
    expected_result = [(scores.references[i][0], scores.scores[i, 2][0], scores.scores[i, 2][1]) for i in range(3)]
```

Reply:
I did try to unpack the tuples. For instance like this:
```python
    def scores_by_reference(self, reference: ReferencesType) -> numpy.ndarray:
        """Return all scores (not sorted) for the given reference spectrum.

        Parameters
        ----------
        reference
            Single reference Spectrum.
        """
        assert reference in self.references, "Given input not found in references."
        selected_idx = int(numpy.where(self.references[:, 0] == reference)[0])
        selected_scores = []
        for i, score in enumerate(self._scores[selected_idx, :].copy()):
            if isinstance(score, tuple):
                selected_scores.append((self.queries[0, i], *score))
            else:
                selected_scores.append((self.queries[0, i], score))
        return selected_scores
```
However that makes it slower (about 2-3 x). Maybe not a big issue

Reply:
But we can also reconsider it later when we work on #143 I guess.

Reply:
Could switch to yielding instead of appending, the slowness will be spread around instead of having to wait a longer time once. 
Agree, lets address that in #143